pcsx2-qt/Debugger/BreakpointDialog.cpp

```diff
@@ -0,0 +1,207 @@
+	m_ui.txtSize->setText(QString::number(m_mc->end - m_mc->start, 16).toUpper());
+	m_ui.chkLog->setChecked(m_mc->result & MEMCHECK_LOG);
+	m_ui.chkEnable->setChecked(m_mc->result & MEMCHECK_BREAK);
+	m_ui.chkChange->setChecked(m_mc->cond & MEMCHECK_WRITE_ONCHANGE);
+	m_ui.chkRead->setChecked(m_mc->cond & MEMCHECK_READ);
+	m_ui.chkWrite->setChecked(m_mc->cond & MEMCHECK_WRITE);
+
+	connect(m_ui.rdoExecute, &QRadioButton::toggled, this, &BreakpointDialog::onRdoButtonToggled);
+	connect(m_ui.rdoMemory, &QRadioButton::toggled, this, &BreakpointDialog::onRdoButtonToggled);
+	m_ui.rdoMemory->toggle();
+}
+
+BreakpointDialog::~BreakpointDialog()
+{
+}
+
+void BreakpointDialog::onRdoButtonToggled()
+{
+	const bool isExecute = m_ui.rdoExecute->isChecked();
+
+	m_ui.grpExecute->setEnabled(isExecute);
+	m_ui.grpMemory->setEnabled(!isExecute);
+
+	m_ui.chkLog->setEnabled(!isExecute);
+}
+
+// When we are creating a breakpoint, use m_bp or m_mc only as a place to store information for the new breakpoint
+// When we are modifying a breakpoint, m_bp or m_mc is a pointer to the breakpoint we are updating
+void BreakpointDialog::accept()
+{
+	if (m_purpose == BPDIALOG_PURPOSE::CREATE)
+	{
+		if (m_ui.rdoExecute->isChecked())
+		{
+			m_bp = new BreakPoint;
+			m_bp->cpu = m_cpu->getCpuType();
+			m_bp->cond.debug = m_cpu;
+		}
+		else
+		{
+			m_mc = new MemCheck;
+			m_mc->cpu = m_cpu->getCpuType();
+		}
+	}
+
+	PostfixExpression expression;
+
+	// Validate the address
+	u64 address;
+	if (!m_cpu->initExpression(m_ui.txtAddress->text().toLocal8Bit(), expression) || !m_cpu->parseExpression(expression, address))
+	{
+		QMessageBox::warning(this, "Error", "Invalid address \"" + m_ui.txtAddress->text() + "\"");
+		return;
+	}
+
+	u64 size;
+	if (m_ui.rdoMemory->isChecked())
+	{
+		if (!m_cpu->initExpression(m_ui.txtSize->text().toLocal8Bit(), expression) || !m_cpu->parseExpression(expression, size) || !size)
+		{
+			QMessageBox::warning(this, "Error", "Invalid size \"" + m_ui.txtSize->text() + "\"");
+			return;
+		}
+
+		m_mc->start = address;
+		const bool changedSize = m_mc->end != (m_mc->start + size);
+		const u32 prevEnd = m_mc->end;
+		m_mc->end = m_mc->start + size;
+
+		int condition = 0;
+		if (m_ui.chkRead->isChecked())
+			condition |= MEMCHECK_READ;
+		if (m_ui.chkWrite->isChecked())
+			condition |= MEMCHECK_WRITE;
+		if (m_ui.chkChange->isChecked())
+			condition |= MEMCHECK_WRITE_ONCHANGE;
+
+		int result = 0;
+		if (m_ui.chkEnable->isChecked())
+			result |= MEMCHECK_BREAK;
+		if (m_ui.chkLog->isChecked())
+			result |= MEMCHECK_LOG;
+
+		if (m_purpose == BPDIALOG_PURPOSE::CREATE)
+		{
+			Host::RunOnCPUThread([this, condition, result] {
+				CBreakPoints::AddMemCheck(m_cpu->getCpuType(), m_mc->start, m_mc->end, (MemCheckCondition)condition, (MemCheckResult)result);
+				delete m_mc;
+			});
+		}
+		else
+		{
+			const auto m_mc_copy = *m_mc;
+			Host::RunOnCPUThread([this, m_mc_copy, condition, result, prevEnd, changedSize] {
```

Comment:
Don't need to worry about changing it here, but for future reference, you can put this in the capture itself, i.e.
```
Host::RunOnCPUThread([this, mc = *m_mc, condition, result, prevEnd, changedSize] {
```